Parser/lexer/lexer.c

```diff
@@ -1317,15 +1329,18 @@ tok_get_fstring_mode(struct tok_state *tok, tokenizer_mode* current_tok, struct
     tok->multi_line_start = tok->line_start;
     while (end_quote_size != current_tok->f_string_quote_size) {
         int c = tok_nextc(tok);
-        if (tok->done == E_ERROR) {
+        if (tok->done == E_ERROR || tok->done == E_DECODE) {
```

Comment:
This was a bug, as if we enter `E_DECODE` state we were not returning soon enough

Reply:
Recently came across this in another branch I'm working on. If we check for this here, we can probably remove the `if (tok->decoding_erred)` about 10 lines below, right?

Reply:
Good idea, will do